user.py

```diff
@@ -90,9 +90,19 @@ def users(user_type):
     return render_template('users.html', users=all_users, user_type=user_type)
 
 
+def is_allowed_user(user_id):
+    user = db.session.query(User).filter_by(id=session["user_id"]).first()
+    return user.id != session["user_id"] and not user.allowed(Role.ADMIN)
```

Comment:
```suggestion
    return user_id == session["user_id"] or session["is_admin"]
```

The function name indicates the opposite of what it is meant to do. Your first condition is however always false here. You can actually rely on session information only.